### PR TITLE
BF: Fixes for using custom config

### PR DESCRIPTION
**Pull request by hoechenberger**
We need to read the `bids_root` from the custom configuration early (and not at the very end of `config.py`, as is currently the case in `master`).

------

It's becoming more and more clear to me that `config.py`'s complexity is slowly getting out of hand and, like @jasmainak said, it's drifting away from what an ordinary user should have to deal with. I would suggest to start treating it more like an ordinary Python module than just a collection of configuration settings (because in fact we're already past this point anyway), and move the bits that we want to be "user-facing" to an example of a custom configuration file. This PR and #84 make it easy to use different custom configurations; and I'm considering adding a small script `run_pipeline.py` to replace the `Makefile` and allow the user to specify which configuration to use without having to set an `MNE_BIDS_CONFIG_PATH` einvironment variable.

WDYT?

### CODE REVIEW

**Comment by agramfort** [comment]
@hoechenberger we need to talk :)

**Comment by hoechenberger** [comment]
> @hoechenberger we need to talk :)

Yeeesss... so... how about the three of us -- you, @jasmainak and I -- schedule a call?

**Comment by jasmainak** [comment]
I'm happy to chat, for example tomorrow morning. But I don't want to stand in the way of progress in the `mne-study-template`. If you think it's more efficient to just talk with @agramfort please go ahead!

**Comment by hoechenberger** [comment]
… ugh … enough for today, will fix the tests tomorrow.

@jasmainak I had a quick call with @agramfort this afternoon and we discussed how to proceed with this PR and some small other steps. I think we are good for now in terms of deciding where to go, at least for the next couple days! :)

**Comment by hoechenberger** [comment]
Should pass tests now… These are quite a number of changes, will explain tomorrow.

**Comment by hoechenberger** [comment]
Ok so this PR now adds a few convenience functions to retrieve certain configuration values:

- `get_subjects`
- `get_subjects_dir`
- `get_sessions`
- `get_task`
- `get_runs`
- `get_kind`

This allows for easier handling of those settings in a central place; for example, a user can now specify `sessions = 'all'` to process all sessions, which is probably more intuitive than `sessions = None`. It also allowed us to move reading of the custom configuration file to the bottom of `config.file` again.

@agramfort I think this is good to merge.

File: config.py [hunk 1]
```diff
@@ -788,3 +770,106 @@ def default_reject_comps_factory():
...
+    raise ValueError(msg)
+elif any([ch_type not in ('meg', 'mag', 'grad') for ch_type in ch_types]):
+    msg = ('Invalid channel type passed. Please adjust `ch_types` in your '
+           'configuration.')
+    raise ValueError(msg)
+
+if 'eeg' in ch_types:
+    if use_ssp:
+        msg = ('You requested SSP for EEG data via use_ssp=True. However, '
+               'this does not work. Please ICA instead by setting '
```
Comment (agramfort) [inline config.py:792]:
```suggestion
               'this is not presently supported. Please use ICA instead by setting '
```

File: config.py [hunk 2]
```diff
@@ -788,3 +770,106 @@ def default_reject_comps_factory():
...
+               'use_ssp=False and use_ica=True.')
+        raise ValueError(msg)
+    if not use_ica:
+        msg = ('You did not request ICA artifact correction for your data. '
+               'To turn it on, set use_ica=True.')
+        mne.utils.warn(msg)
```
Comment (agramfort) [inline config.py:798]: a warn suggests that ICA is mandatory for EEG. This is not for me.  I would make it an info message.

Comment (hoechenberger) [reply config.py]: Changed to an ordinary log message in 0141852

**Comment by agramfort** [comment]
thansk a lot @hoechenberger  !

**Pull request merged**